core/services/job/orm.go

```diff
@@ -698,6 +701,42 @@ func (o *orm) FindJob(ctx context.Context, id int32) (jb Job, err error) {
 	return
 }
 
+// FindJobWithoutSpecErrors returns a job by ID, without loading Spec Errors preloaded
+func (o *orm) FindJobWithoutSpecErrors(id int32) (jb Job, err error) {
+	ctx, cancel := pg.DefaultQueryCtx()
+	defer cancel()
+
+	q := o.q.WithOpts(pg.WithParentCtx(ctx))
+	err = q.Transaction(func(tx pg.Queryer) error {
+		stmt := "SELECT * FROM jobs WHERE id = $1 LIMIT 1"
+		err = tx.Get(&jb, stmt, id)
+		if err != nil {
+			return errors.Wrap(err, "failed to load job")
+		}
+
+		if err = LoadAllJobTypes(tx, &jb); err != nil {
+			return errors.Wrap(err, "failed to load job types")
+		}
+
+		return nil
+	})
```

Comment:
This could be a readonly transaction

Reply:
Nice catch, I wasn't aware of the existence of it! Thanks!